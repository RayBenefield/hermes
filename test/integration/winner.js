import flame from '@leonardvandriel/flame';
import * as databases from './database'; // eslint-disable-line import/no-unresolved,import/extensions,no-unused-vars
import uuid from '../utils/mock-uuid';
import random from '../utils/mock-random';
import configureSave from '../utils/save-db';
import configureDomain from '../../src/domain';
import configureFlame from '../../src/lib/local-db';

const db = configureFlame(flame);
const domain = configureDomain({ db, uuid, random });
const save = configureSave(flame); // eslint-disable-line no-unused-vars
beforeEach(() => flame.loadDatabase(undefined));

test('notify all players of the winner and the new round', () => {
    // Given
    flame.loadDatabase(databases['winner-decided']);
    uuid(); // Needed to trigger the next random generation for the new round

    // When
    return domain({
        action: 'winner',
        payload: {
            game: '0-05526803409810505',
            round: '0-05526803409810505',
            winner: 175,
        },
    })
        .then(save('winner-then-new-round'))
        .then(results => expect({
            results,
            db: flame.get('/'),
        }).toMatchSnapshot());
});
