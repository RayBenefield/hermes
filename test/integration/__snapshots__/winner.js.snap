// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notify all players of the winner and the new round 1`] = `
Object {
  "db": Object {
    "candidates": Object {
      "0-05526803409810505": Object {
        "0-05526803409810505": Object {
          "candidates": Object {
            "123456": 268,
            "234567": 188,
            "345678": 175,
            "456789": 117,
            "567890": 31,
          },
          "game": "0-05526803409810505",
          "notified_players": Object {
            "567890": true,
          },
          "round": "0-05526803409810505",
        },
      },
    },
    "games": Object {
      "0-05526803409810505": Object {
        "id": "0-05526803409810505",
        "notified_players": Object {
          "123456": true,
          "234567": true,
          "345678": true,
          "456789": true,
          "567890": true,
        },
        "players": Object {
          "123456": true,
          "234567": true,
          "345678": true,
          "456789": true,
          "567890": true,
        },
        "rounds": Object {
          "0": "0-05526803409810505",
        },
      },
    },
    "hands": Object {
      "0-05526803409810505": Object {
        "123456": Object {
          "cards": Object {
            "1015": true,
            "1039": true,
            "1040": true,
            "310": true,
            "444": true,
            "597": true,
            "640": true,
            "714": true,
            "748": true,
          },
          "game": "0-05526803409810505",
          "player": 123456,
        },
        "234567": Object {
          "cards": Object {
            "218": true,
            "370": true,
            "429": true,
            "485": true,
            "624": true,
            "629": true,
            "725": true,
            "779": true,
            "951": true,
          },
          "game": "0-05526803409810505",
          "player": 234567,
        },
        "345678": Object {
          "cards": Object {
            "252": true,
            "322": true,
            "458": true,
            "582": true,
            "590": true,
            "609": true,
            "900": true,
            "965": true,
            "999": true,
          },
          "game": "0-05526803409810505",
          "player": 345678,
        },
        "456789": Object {
          "cards": Object {
            "241": true,
            "244": true,
            "334": true,
            "470": true,
            "498": true,
            "524": true,
            "593": true,
            "855": true,
            "898": true,
          },
          "game": "0-05526803409810505",
          "player": 456789,
        },
        "567890": Object {
          "cards": Object {
            "125": true,
            "227": true,
            "32": true,
            "36": true,
            "530": true,
            "597": true,
            "604": true,
            "605": true,
            "729": true,
          },
          "game": "0-05526803409810505",
          "player": 567890,
        },
      },
    },
    "players": Object {
      "facebook": Object {
        "123456": Object {
          "first_name": "Ray",
          "games": Object {
            "0-05526803409810505": true,
          },
          "id": 123456,
          "platform": "facebook",
        },
        "234567": Object {
          "first_name": "Hermon",
          "games": Object {
            "0-05526803409810505": true,
          },
          "id": 234567,
          "platform": "facebook",
        },
        "345678": Object {
          "first_name": "Hermes",
          "games": Object {
            "0-05526803409810505": true,
          },
          "id": 345678,
          "platform": "facebook",
        },
        "456789": Object {
          "first_name": "Hermshire",
          "games": Object {
            "0-05526803409810505": true,
          },
          "id": 456789,
          "platform": "facebook",
        },
        "567890": Object {
          "first_name": "Hermda",
          "games": Object {
            "0-05526803409810505": true,
          },
          "id": 567890,
          "platform": "facebook",
        },
      },
    },
    "queue": Object {},
    "rounds": Object {
      "0-05526803409810505": Object {
        "0-05526803409810505": Object {
          "candidates": Object {
            "123456": 268,
            "234567": 188,
            "345678": 175,
            "456789": 117,
            "567890": 31,
          },
          "card": Object {
            "contents": "I can only express myself through _.",
            "id": 189,
          },
          "game": "0-05526803409810505",
          "id": "0-05526803409810505",
          "notified_players": Object {
            "123456": true,
            "234567": true,
            "345678": true,
            "456789": true,
            "567890": true,
          },
          "votes": Object {
            "123456": Object {
              "0": "345678",
              "1": "456789",
              "2": "567890",
            },
            "234567": Object {
              "0": "345678",
              "1": "456789",
              "2": "567890",
            },
            "345678": Object {
              "0": "123456",
              "1": "456789",
              "2": "567890",
            },
            "456789": Object {
              "0": "345678",
              "1": "123456",
              "2": "567890",
            },
            "567890": Object {
              "0": "345678",
              "1": "456789",
              "2": "123456",
            },
          },
          "winner": 175,
        },
        "0-9057442687334262": Object {
          "game": "0-05526803409810505",
          "id": "0-9057442687334262",
        },
      },
    },
  },
  "results": Object {
    "action": "winner",
    "candidates": Object {
      "123456": Object {
        "contents": "Dropping a chandelier on your enemies and riding the rope up.",
        "id": 268,
      },
      "234567": Object {
        "contents": "Keg stands.",
        "id": 188,
      },
      "345678": Object {
        "contents": "Skeletor.",
        "id": 175,
      },
      "456789": Object {
        "contents": "Sniffing glue.",
        "id": 117,
      },
      "567890": Object {
        "contents": "The forbidden fruit.",
        "id": 31,
      },
    },
    "game": Object {
      "id": "0-05526803409810505",
      "notified_players": Object {
        "123456": true,
        "234567": true,
        "345678": true,
        "456789": true,
        "567890": true,
      },
      "players": Object {
        "123456": true,
        "234567": true,
        "345678": true,
        "456789": true,
        "567890": true,
      },
      "rounds": Object {
        "0": "0-05526803409810505",
      },
    },
    "messages": Object {
      "123456": Array [
        Object {
          "payload": Object {
            "goal": Object {
              "contents": "I can only express myself through _.",
              "id": 189,
            },
            "player": Object {
              "first_name": "Hermes",
              "games": Object {
                "0-05526803409810505": true,
              },
              "id": 345678,
              "platform": "facebook",
            },
            "winner": Object {
              "contents": "Skeletor.",
              "id": 175,
            },
          },
          "player": 123456,
          "type": "notify-winner-message",
        },
      ],
      "234567": Array [
        Object {
          "payload": Object {
            "goal": Object {
              "contents": "I can only express myself through _.",
              "id": 189,
            },
            "player": Object {
              "first_name": "Hermes",
              "games": Object {
                "0-05526803409810505": true,
              },
              "id": 345678,
              "platform": "facebook",
            },
            "winner": Object {
              "contents": "Skeletor.",
              "id": 175,
            },
          },
          "player": 234567,
          "type": "notify-winner-message",
        },
      ],
      "345678": Array [
        Object {
          "payload": Object {
            "goal": Object {
              "contents": "I can only express myself through _.",
              "id": 189,
            },
            "player": Object {
              "first_name": "Hermes",
              "games": Object {
                "0-05526803409810505": true,
              },
              "id": 345678,
              "platform": "facebook",
            },
            "winner": Object {
              "contents": "Skeletor.",
              "id": 175,
            },
          },
          "player": 345678,
          "type": "notify-winner-message",
        },
      ],
      "456789": Array [
        Object {
          "payload": Object {
            "goal": Object {
              "contents": "I can only express myself through _.",
              "id": 189,
            },
            "player": Object {
              "first_name": "Hermes",
              "games": Object {
                "0-05526803409810505": true,
              },
              "id": 345678,
              "platform": "facebook",
            },
            "winner": Object {
              "contents": "Skeletor.",
              "id": 175,
            },
          },
          "player": 456789,
          "type": "notify-winner-message",
        },
      ],
      "567890": Array [
        Object {
          "payload": Object {
            "goal": Object {
              "contents": "I can only express myself through _.",
              "id": 189,
            },
            "player": Object {
              "first_name": "Hermes",
              "games": Object {
                "0-05526803409810505": true,
              },
              "id": 345678,
              "platform": "facebook",
            },
            "winner": Object {
              "contents": "Skeletor.",
              "id": 175,
            },
          },
          "player": 567890,
          "type": "notify-winner-message",
        },
      ],
    },
    "payload": Object {
      "game": "0-05526803409810505",
      "round": "0-05526803409810505",
      "winner": 175,
    },
    "players": Array [
      Object {
        "first_name": "Ray",
        "games": Object {
          "0-05526803409810505": true,
        },
        "id": 123456,
        "platform": "facebook",
      },
      Object {
        "first_name": "Hermon",
        "games": Object {
          "0-05526803409810505": true,
        },
        "id": 234567,
        "platform": "facebook",
      },
      Object {
        "first_name": "Hermes",
        "games": Object {
          "0-05526803409810505": true,
        },
        "id": 345678,
        "platform": "facebook",
      },
      Object {
        "first_name": "Hermshire",
        "games": Object {
          "0-05526803409810505": true,
        },
        "id": 456789,
        "platform": "facebook",
      },
      Object {
        "first_name": "Hermda",
        "games": Object {
          "0-05526803409810505": true,
        },
        "id": 567890,
        "platform": "facebook",
      },
    ],
    "round": Object {
      "candidates": Object {
        "123456": 268,
        "234567": 188,
        "345678": 175,
        "456789": 117,
        "567890": 31,
      },
      "card": Object {
        "contents": "I can only express myself through _.",
        "id": 189,
      },
      "game": "0-05526803409810505",
      "id": "0-05526803409810505",
      "notified_players": Object {
        "567890": true,
      },
      "votes": Object {
        "123456": Object {
          "0": "345678",
          "1": "456789",
          "2": "567890",
        },
        "234567": Object {
          "0": "345678",
          "1": "456789",
          "2": "567890",
        },
        "345678": Object {
          "0": "123456",
          "1": "456789",
          "2": "567890",
        },
        "456789": Object {
          "0": "345678",
          "1": "123456",
          "2": "567890",
        },
        "567890": Object {
          "0": "345678",
          "1": "456789",
          "2": "123456",
        },
      },
      "winner": 175,
    },
    "unaddressedMessages": Array [
      Object {
        "payload": Object {
          "goal": Object {
            "contents": "I can only express myself through _.",
            "id": 189,
          },
          "player": Object {
            "first_name": "Hermes",
            "games": Object {
              "0-05526803409810505": true,
            },
            "id": 345678,
            "platform": "facebook",
          },
          "winner": Object {
            "contents": "Skeletor.",
            "id": 175,
          },
        },
        "player": 123456,
        "type": "notify-winner-message",
      },
      Object {
        "payload": Object {
          "goal": Object {
            "contents": "I can only express myself through _.",
            "id": 189,
          },
          "player": Object {
            "first_name": "Hermes",
            "games": Object {
              "0-05526803409810505": true,
            },
            "id": 345678,
            "platform": "facebook",
          },
          "winner": Object {
            "contents": "Skeletor.",
            "id": 175,
          },
        },
        "player": 234567,
        "type": "notify-winner-message",
      },
      Object {
        "payload": Object {
          "goal": Object {
            "contents": "I can only express myself through _.",
            "id": 189,
          },
          "player": Object {
            "first_name": "Hermes",
            "games": Object {
              "0-05526803409810505": true,
            },
            "id": 345678,
            "platform": "facebook",
          },
          "winner": Object {
            "contents": "Skeletor.",
            "id": 175,
          },
        },
        "player": 345678,
        "type": "notify-winner-message",
      },
      Object {
        "payload": Object {
          "goal": Object {
            "contents": "I can only express myself through _.",
            "id": 189,
          },
          "player": Object {
            "first_name": "Hermes",
            "games": Object {
              "0-05526803409810505": true,
            },
            "id": 345678,
            "platform": "facebook",
          },
          "winner": Object {
            "contents": "Skeletor.",
            "id": 175,
          },
        },
        "player": 456789,
        "type": "notify-winner-message",
      },
      Object {
        "payload": Object {
          "goal": Object {
            "contents": "I can only express myself through _.",
            "id": 189,
          },
          "player": Object {
            "first_name": "Hermes",
            "games": Object {
              "0-05526803409810505": true,
            },
            "id": 345678,
            "platform": "facebook",
          },
          "winner": Object {
            "contents": "Skeletor.",
            "id": 175,
          },
        },
        "player": 567890,
        "type": "notify-winner-message",
      },
    ],
    "unnotifiedPlayers": Array [
      Object {
        "first_name": "Ray",
        "games": Object {
          "0-05526803409810505": true,
        },
        "id": 123456,
        "platform": "facebook",
      },
      Object {
        "first_name": "Hermon",
        "games": Object {
          "0-05526803409810505": true,
        },
        "id": 234567,
        "platform": "facebook",
      },
      Object {
        "first_name": "Hermes",
        "games": Object {
          "0-05526803409810505": true,
        },
        "id": 345678,
        "platform": "facebook",
      },
      Object {
        "first_name": "Hermshire",
        "games": Object {
          "0-05526803409810505": true,
        },
        "id": 456789,
        "platform": "facebook",
      },
      Object {
        "first_name": "Hermda",
        "games": Object {
          "0-05526803409810505": true,
        },
        "id": 567890,
        "platform": "facebook",
      },
    ],
    "winner": Object {
      "contents": "Skeletor.",
      "id": 175,
    },
    "winningPlayer": Object {
      "first_name": "Hermes",
      "games": Object {
        "0-05526803409810505": true,
      },
      "id": 345678,
      "platform": "facebook",
    },
  },
}
`;
